tests/template_tests/test_partials.py

```diff
@@ -0,0 +1,303 @@
+
+    def test_undefined_partial_error_message(self):
+
+        template = """
+        {% partialdef different-partial %}
+        THIS IS THE NOT DEFINED PARTIAL CONTENT
+        {% endpartialdef %}
+        {% partial not-defined-partial %}
+        """
+
+        engine = engines["django"]
+        t = engine.from_string(template)
+        with self.assertRaisesMessage(
+            TemplateSyntaxError,
+            "You are trying to access an undefined partial 'not-defined-partial'",
+        ):
+            t.render({})
+
+    def test_undefined_partial_error_message_when_no_partialdef(self):
+
+        template = """
+        {% partial not-defined-partial %}
+        """
+        engine = engines["django"]
+        t = engine.from_string(template)
+        with self.assertRaisesMessage(
+            TemplateSyntaxError,
+            "No partials are defined. You are trying to access "
+            "'not-defined-partial' partial",
+        ):
+            t.render({})
+
+
+class PartialTagsCacheTestCase(TestCase):
+    def test_partials_use_cached_loader_when_configured(self):
+
+        template_dir = os.path.join(os.path.dirname(__file__), "templates")
+        backend = DjangoTemplates(
+            {
+                "NAME": "django",
+                "DIRS": [template_dir],
+                "APP_DIRS": False,
+                "OPTIONS": {
+                    "loaders": [
+                        (
+                            "django.template.loaders.cached.Loader",
+                            ["django.template.loaders.filesystem.Loader"],
+                        ),
+                    ],
+                },
+            }
+        )
+
+        cached_loader = backend.engine.template_loaders[0]
+        filesystem_loader = cached_loader.loaders[0]
+
+        with mock.patch.object(
+            filesystem_loader, "get_contents", wraps=filesystem_loader.get_contents
+        ) as mock_get_contents:
+            full_template = backend.get_template("partial_examples.html")
+            rendered_full = full_template.render({})
+            self.assertIn("TEST-PARTIAL-CONTENT", rendered_full)
+
+            partial_template = backend.get_template(
+                "partial_examples.html#test-partial"
+            )
+            rendered_partial = partial_template.render({})
+            self.assertEqual("TEST-PARTIAL-CONTENT", rendered_partial.strip())
+
+            mock_get_contents.assert_called_once()
+
+
+class ResponseContextWithPartialTests(TestCase):
+
+    def test_response_context_available_for_partial_template(self):
+        def sample_view(request):
+
+            return HttpResponse(
+                render_to_string("partial_examples.html#test-partial", {"foo": "bar"})
+            )
+
+        urls_module = ModuleType("partial_test_urls")
+        urls_module.urlpatterns = [path("sample/", sample_view, name="sample-view")]
+
+        with override_settings(
+            ROOT_URLCONF=urls_module,
+        ):
```

Comment:
This wrapping is not needed, could you please double check the code to ensure lines shorter that 89cols fit in one line?
```suggestion
        with override_settings(ROOT_URLCONF=urls_module):
```